cli/cmd/main_test.go

```diff
@@ -16,13 +16,17 @@ var (
 
 	// prettyDiff is set by the `-pretty-diff` flag.
 	prettyDiff bool
+
+	// write any rejected test data into this path
+	rejectPath string
 )
 
 // TestMain parses flags before running tests
 func TestMain(m *testing.M) {
 	flag.BoolVar(&updateFixtures, "update", false, "update text fixtures in place")
 	prettyDiff = os.Getenv("LINKERD_TEST_PRETTY_DIFF") != ""
 	flag.BoolVar(&prettyDiff, "pretty-diff", prettyDiff, "display the full text when diffing")
+	flag.StringVar(&rejectPath, "reject-path", "", "write results for failed tests to this path (path is relative to the test location)")
```

Comment:
Can you explain the benefits of this new functionality, compared to the current diffs shown in failures? Since this is separate than what the current PR addresses, I would prefer this to be submitted in a different PR.

Reply:
For failed tests, it writes the generated resource definitions to a given path. This makes it possible to compare the golden file and the test result with more powerful diff tools than the pretty print.

Background: When debugging certs, the diff often is in the middle of long strings of line noise. Neither the normal nor the pretty diff output are well suited for finding the reason for the difference, as the length and type of the diff will push out the context text out of the visible scope.

Reply:
Ah that's pretty neat :+1: